src/main/java/net/mcreator/blockly/java/blocks/CustomDependencyBlock.java

```diff
@@ -60,8 +52,11 @@
 	}
 
 	@Override public String[] getSupportedBlocks() {
-		return new String[] { "custom_dependency_logic", "custom_dependency_number", "custom_dependency_text",
-				"custom_dependency_itemstack" };
+		Set<String> names = new HashSet<>();

```

Comment:
These shold be cached, move this to object constructor. Applies to all blocks that do this.